### DISCUSSION TITLE
Getting URL to View on Github Pages

### DISCUSSION THREAD

**Discussion by Benjamin-bm** [community/community#58879] [New to GitHub]
Hi
I always almost everytime have a problem to get the URL to  View on Github Pages for my project. Could somebody help me out? I'm missing something or did something wrong.
This is the project:
https://github.com/Benjamin-bm/Capstone.project/tree/main/Capstone.project

I have tried using this guide:https://tomcam.github.io/least-github-pages/github-pages-url.html
And come up with: https://benjamin-bm.github.io/Capstone.project/Capstone.project/

Best,

Benjamin

**Comment by Gustavonn07** [comment]
I don't know if you used git hub pages, but it is used to get a web page from your repository

Here are some steps you can follow to address the issue:

1. Check repository settings: Make sure the correct branch is selected for publishing and that the deployment folder is set up correctly. You mentioned that you've already checked this, but it's important to review it again to ensure the settings are correct.

2. Verify folder and file structure: Check if the necessary HTML, CSS, JavaScript, and other required files are present in the correct folder in your repository. Ensure that the folder structure is correct and that the file links are properly defined in the HTML files.

3. Verify GitHub Pages is enabled: Go to your repository settings on GitHub and scroll down to the "GitHub Pages" section. Check if the "GitHub Pages" option is enabled and if the branch and deployment folder are set correctly.

4. Check for code errors: Verify if there are any errors in your HTML, CSS, or JavaScript code. Syntax errors or missing references can cause issues with page rendering. Use the developer tools in your browser to check for errors and fix them if needed.

5. Clear browser cache: Sometimes, pages may not display correctly due to browser cache. Try clearing your browser cache or open the pages in a different browser to see if it resolves the issue.

6. Check for permission issues: Ensure that the permissions for your repository are set correctly. If you're using a private repository, check if the people who need access to the pages have the correct permissions.

I don't know how you use github pages, but I use it this way and it rarely gives me a problem:
![image](https://github.com/community/community/assets/84361085/1a76af41-5924-4847-a435-b3985ca6465a)

<hr>

If the main HTML file is not named "index.html", it can cause problems when viewing GitHub pages. GitHub Pages expects the main file to be named "index.html" by default. When you access the GitHub Pages URL (for example, https://yourusername.github.io/), the server looks for a file named "index.html" in the specified deployment folder. If the main file has another name, the server will not be able to find it automatically and the pages will not be displayed correctly.

So, to ensure that your GitHub pages work correctly, make sure your main HTML file is named "index.html". If the file name is different, you can either rename it to "index.html" or configure GitHub Pages to explicitly point to the correct file name in your repository settings.

i hope it helps!

**Comment by MonteWest** [comment]
Any Idea how I configure GitHub Pages to explicitly point to the correct file name in your repository settings?
